src/Cli/dotnet/commands/dotnet-workload/SignCheck.cs

```diff
@@ -2,34 +2,57 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Reflection;
+#if !DOT_NET_BUILD_FROM_SOURCE
 using Microsoft.DotNet.Installer.Windows.Security;
+#endif
 using Microsoft.Win32;
 
 namespace Microsoft.DotNet.Workloads.Workload
 {
     internal static class SignCheck
     {
+        internal static readonly string OnlineRevocationCheckPolicyKeyName = "AllowOnlineRevocationChecks";
+        internal static readonly string VerifySignaturesPolicyKeyName = "VerifySignatures";
+
+        private static readonly string s_WorkloadPolicyKey = @"SOFTWARE\Policies\Microsoft\dotnet\Workloads";
+
         private static readonly string s_dotnet = Assembly.GetExecutingAssembly().Location;
 
         /// <summary>
-        /// Determines whether dotnet is signed.
+        /// Determines whether dotnet.dll is signed.
         /// </summary>
-        /// <returns><see langword="true"/> if dotnet is signed; <see langword="false"/> otherwise.</returns>
-        public static bool IsDotNetSigned() => IsSigned(s_dotnet);
+        /// <returns><see langword="true"/> if dotnet is signed; otherwise, <see langword="false"/>.</returns>
+        public static bool IsDotNetSigned()
+        {
+            if (OperatingSystem.IsWindows())
+            {
+#if !DOT_NET_BUILD_FROM_SOURCE
```

Comment:
Does this mean we don't check the authenticode signature for source builds?

Reply:
yup, SB gets unhappy - we don't ship SB copies.